### PR TITLE
Basic Login Functionality

### PR DESCRIPTION
**Pull request by hahnspena102**
These changes implement the login, register, and sign out. DEFINITELY DOESN'T LOOK GOOD (but it works). Will make the frontend prettier in the future.

### CODE REVIEW

File: packages/express-backend/auth.js [hunk 1]
```diff
@@ -1,28 +1,37 @@
...
 const creds = [];
 
 export function registerUser(req, res) {
     const { username, password } = req.body; // from form
-  
+    console.log(creds);
+
     if (!username || !password) {
-      res.status(400).send("Bad request: Invalid input data.");
-    } else if (creds.find((c) => c.username === username)) {
```
Comment (ishwaryanam) [inline packages/express-backend/auth.js:12]: don't forgot to remove creds!

**Review by ishwaryanam — COMMENTED** [review]
looks good to me! good luck as you continue to work on this

File: packages/react-frontend/src/pages/Login.jsx [hunk 2]
```diff
@@ -0,0 +1,135 @@
...
+}
+
+function LoginPage() {
+  const INVALID_TOKEN = "INVALID_TOKEN";
+  const [token, setToken] = useState(localStorage.getItem("authToken") || INVALID_TOKEN);
+  //const [token, setToken] = useState(INVALID_TOKEN);
+  const [message, setMessage] = useState("");
+  const [isAuthenticated, setIsAuthenticated] = useState(false);
+
+  const API_PREFIX = "http://localhost:8000";
```
Comment (ArjunR27) [inline packages/react-frontend/src/pages/Login.jsx:63]: We are going to have to change this to
const API_PREFIX = https://songsmith.azurewebsites.net

File: packages/react-frontend/src/pages/Signup.jsx [hunk 3]
```diff
@@ -1,10 +1,99 @@
...
+
 
 export default function Signup() {
+  const INVALID_TOKEN = "INVALID_TOKEN";
+  const [token, setToken] = useState(INVALID_TOKEN);
+  const [message, setMessage] = useState("");
+  const [isAuthenticated, setIsAuthenticated] = useState(false);
+
+  function signupUser(creds) {
+    const API_PREFIX = "http://localhost:8000";
```
Comment (ArjunR27) [inline packages/react-frontend/src/pages/Signup.jsx:66]: Same thing here as well
const API_PREFIX = "https://songsmith.azurewebsites.net"

**Review by ArjunR27 — COMMENTED** [review]
Overall, looks good to me. Going to need to add some minor changes now with CI/CD up and running

**Comment by poornimagodavarthy2** [comment]
Looks good overall, good luck as you continue to work on it!

**Pull request merged**